### DISCUSSION TITLE
Is it possible to see/lookup the commands that are issued when I click some action in cockpit?

### DISCUSSION THREAD

**Discussion by Tronde** [cockpit-project/cockpit#18716] [Q&A]
Hi folks,  
Here is my example that should help understand my question:

I'm using the `cockpit-machines` plugin to manage my virtual machines (VM). When I add some disk to a VM I would like to know which exactly command was used including it's arguments, parameters and options.

Is there any log where I could find this information?

For some more background on my question please see [my issue here](https://github.com/Tronde/kvm_provision_lab/issues/1).

Thank you in advance for you help and guidance.

**Comment by martinpitt** [comment]
A lot of the actions that Cockpit (also the Machines page) do is not through executing commands, but through other APIs -- most of them are through [D-Bus services](https://en.wikipedia.org/wiki/D-Bus), some also through writing files or sending socket requests. That said, cockpit-machines calls `virt-xml` and `virt-install` a lot, and for your issue that could actually be enough. You can see what happens by opening the developer console (Ctrl+Shift+J in Chromium, Ctrl+Shift+K in firefox) and setting `window.debugging = "spawn"`. Then any action that you do from then on that causes a program to be called will be logged there, and you also see its output and result.

**Reply by Tronde** [reply]
Where do you set `window.debugging = "spawn"`? Do you set it as filter in the developer console or somewhere else?

**Reply by martinpitt** [reply]
As a JS command in the developer console. But I just saw that current version only gives you "process closed" and "process output", not the actual command argv vector. That needs to be fixed :cry:
